src/App/Fossa/SBOM/Analyze.hs

```diff
@@ -0,0 +1,93 @@
+import Control.Carrier.Debug (Debug)
+import Control.Carrier.Diagnostics qualified as Diag
+import Control.Carrier.FossaApiClient (runFossaApiClient)
+import Control.Carrier.StickyLogger (StickyLogger, logSticky, runStickyLogger)
+import Control.Carrier.Telemetry.Types (CountableCliFeature (SBOMAnalyzeUsage))
+import Control.Effect.FossaApiClient (FossaApiClient, PackageRevision (PackageRevision), getOrganization, getSignedUploadUrl, queueSBOMBuild, uploadArchive)
+import Control.Effect.Lift
+import Control.Effect.Telemetry (Telemetry, trackUsage)
+import Data.Foldable (traverse_)
+import Data.String.Conversion
+import Data.Text (Text)
+import Effect.Logger (Logger, logDebug, logInfo)
+import Fossa.API.Types
+import Prettyprinter (Pretty (pretty))
+import Srclib.Types (Locator (..))
+
+analyze ::
+  ( Has Diag.Diagnostics sig m
+  , Has (Lift IO) sig m
+  , Has Logger sig m
+  , Has Debug sig m
+  , Has Telemetry sig m
+  ) =>
+  SBOMAnalyzeConfig ->
+  m ()
+analyze config = do
+  let emptyMetadata = ProjectMetadata Nothing Nothing Nothing Nothing Nothing Nothing [] Nothing
+  let apiOpts = sbomApiOpts config
+  trackUsage SBOMAnalyzeUsage
+  runFossaApiClient apiOpts . preflightChecks $ AnalyzeChecks (sbomRevision config) emptyMetadata
+  runFossaApiClient apiOpts . runStickyLogger (severity config) $ analyzeInternal config
+
+analyzeInternal ::
+  ( Has Diag.Diagnostics sig m
+  , Has StickyLogger sig m
+  , Has Logger sig m
+  , Has FossaApiClient sig m
+  ) =>
+  SBOMAnalyzeConfig ->
+  m ()
+analyzeInternal config = do
+  let revision = sbomRevision config
+  _ <- uploadSBOM config
+
+  let archive = Archive (projectName revision) (projectRevision revision)
+  _ <- queueSBOMBuild archive (sbomTeam config) (sbomRebuild config)
+
+  -- The locator used in the URL has the organization ID on it, so we
+  -- need to generate that locator before displaying it
+  orgId <- organizationId <$> getOrganization
+  let updateRevisionName :: Text -> ProjectRevision -> ProjectRevision
+      updateRevisionName updateText r = r{projectName = updateText <> "/" <> projectName r}
+      revisionWithOrganization = updateRevisionName (toText $ show orgId) revision
+      locator = Locator "sbom" (projectName revisionWithOrganization) (Just $ projectRevision revision)
+  logSticky $ "uploaded to " <> toText locator
+  buildUrl <- getFossaBuildUrl revisionWithOrganization locator
+
+  traverse_
+    logInfo
+    [ "============================================================"
+    , ""
+    , "    View FOSSA Report:"
+    , "    " <> pretty buildUrl
+    , ""
+    , "============================================================"
+    ]
+
+uploadSBOM ::
+  ( Has StickyLogger sig m
+  , Has FossaApiClient sig m
+  , Has Logger sig m
+  ) =>
+  SBOMAnalyzeConfig ->
+  m ()
+uploadSBOM config = do
+  let revision = sbomRevision config
+  signedURL <- getSignedUploadUrl SBOMUpload $ PackageRevision (projectName revision) (projectRevision revision)
```

Comment:
[optional] It seems like there's a function from `ProjectRevision -> PackageRevision` that would be useful to define. I see this pattern pop up in a few places.